src/Auth0.AuthenticationApi/AuthenticationApiClient.cs

```diff
@@ -211,134 +196,118 @@ public async Task<AccessTokenResponse> GetTokenAsync(AuthorizationCodePkceTokenR
         /// Request an Access Token using the Client Credentials Grant flow.
         /// </summary>
         /// <param name="request">The <see cref="ClientCredentialsTokenRequest"/> containing the information of the request.</param>
-        /// <returns>An <see cref="AccessTokenResponse"/> containing the token information</returns>
+        /// <returns><see cref="AccessTokenResponse"/> containing the desired token information.</returns>
         public Task<AccessTokenResponse> GetTokenAsync(ClientCredentialsTokenRequest request)
         {
-            return Connection.PostAsync<AccessTokenResponse>("oauth/token", null, new Dictionary<string, object>
-                {
-                    {"grant_type", "client_credentials"},
-                    {"client_id", request.ClientId},
-                    {"client_secret", request.ClientSecret},
-                    {"audience", request.Audience}
-                },
-                null,
-                null,
-                null,
-                null);
+            var body = new Dictionary<string, string> {
+                { "grant_type", "client_credentials" },
+                { "client_id", request.ClientId },
+                { "client_secret", request.ClientSecret },
+                { "audience", request.Audience } };
+
+            return connection.SendAsync<AccessTokenResponse>(
+                HttpMethod.Post,
+                tokenUri,
+                body);
         }
 
         /// <summary>
-        /// Given a <see cref="RefreshTokenRequest"/>, it will retrieve a refreshed access token from the authorization server.
+        /// Refresh all tokens using the Refresh Token obtained during authorization.
         /// </summary>
-        /// <param name="request">The refresh token request details, containing a valid refresh token.</param>
-        /// <returns>The new token issued by the server.</returns>
+        /// <param name="request"><see cref="RefreshTokenRequest"/> containing Refresh Token and associated parameters.</param>
+        /// <returns><see cref="AccessTokenResponse"/> containing the desired token information.</returns>
         public async Task<AccessTokenResponse> GetTokenAsync(RefreshTokenRequest request)
         {
-            var parameters = new Dictionary<string, object> {
+            var body = new Dictionary<string, string> {
                 { "grant_type", "refresh_token" },
-                { "refresh_token", request.RefreshToken },
                 { "client_id", request.ClientId },
-                { "client_secret", request.ClientSecret }
+                { "client_secret", request.ClientSecret },
```

Comment:
You may want to make this optional. If this library uses PKCE, that's an indication it might not be storing/using a secret. Refresh tokens can be exchanged without one (if the application is setup to do that).

Reply:
So just add client_secret conditionally?

Reply:
Yes

Reply:
Let's do this and the removal of the unnecessary refresh token parameters in a separate PR to keep the concerns separated.  I'll put it out this morning once this one is in.